cli/azd/pkg/templates/source_manager.go

```diff
@@ -66,14 +68,15 @@ type SourceManager interface {
 	// Remove removes a template source.
 	Remove(ctx context.Context, name string) error
 	// CreateSource creates a new template source from a source configuration
-	CreateSource(ctx context.Context, source *SourceConfig) (Source, error)
+	CreateSource(ctx context.Context, source *SourceConfig, console input.Console) (Source, error)
 }
 
 type sourceManager struct {
 	options        *SourceOptions
 	serviceLocator ioc.ServiceLocator
 	configManager  config.UserConfigManager
 	transport      policy.Transporter
+	ghCli          *github.Cli
```

Comment:
Not all sources are going to need the github CLI reference.  Lets think of some ways where we don't need to take this dependency at the sourceManager level.

Reply:
It's a little complicated when the design is using a top abstraction (the manager) for all implementations.

would it be fine to just make this either Lazy[ghCli] or use a callback to only inject the create-callback with the container but do not create the ghCli unless the manager uses the gh source?